### PR TITLE
Fix cors issue and disable proxy when under https

### PR DESCRIPTION
**Pull request by ivmarcos**
- Resolving issue with browser identification using window instead of process
- Disabling proxy when browser is under https

### CODE REVIEW

File: src/services/index.js [hunk 1]
```diff
@@ -2,12 +2,25 @@ import CepAberto from './cepaberto'
 import Correios from './correios'
 import ViaCep from './viacep'
 import { PROXY_URL } from '../utils/consts'
+import { type } from 'os';
```
[the remark this answers is not in the document]
Comment (ivmarcos) [reply src/services/index.js]: Sorry @mccraveiro, I did not notice it, probably was automatically included. I just pushed an update with the removal.

**Comment by lucianopf** [comment]
Nice catch buddy! (I've studied to use this approach but I don't remember exactly why I haven't done so, probably a edge case where Node has window defined, but that's definitely more an edge case than the browser one)

I've got 2 questions:
1 - Why to remove the proxy when it's under https? That's basically disabling the proxy since that's almost the default protocol used and the proxy also uses https 🤔 
2 - Can you please check what's missing at the coverage check to achieve 100%? It's a mandatory requirement to approve the PR 😬 

Btw, great job and thanks for this PR! ❤️

**Comment by ivmarcos** [comment]
Thanks @lucianopf!  Answering your questions:

1. Why is the proxy required for, if the browser is already using https? Unless the proxy host has its address authorized in the cep webservice as a cors rule, is that the case? Avoiding its use when unnecessary improves latency.
2. Resolved using the same ignore istanbul tags placed on ```injectProxy```.

**Comment by ivmarcos** [comment]
The latency can be minimal and not worrying, but logically is faster using a direct request than using a request by a proxy. Cors headers like ```Allow-Control-Origin``` are already handled by the services in this lib in the fetch functions, so the user already does not need to worry about it, whether or not using a proxy, so I still do not understand why it's better or required to use proxy when https is already being used.

**Comment by mathmarques** [comment]
@ivmarcos The proxy is required to ignore the CORS policy of the providers. The browser won't allow cross-domain XHR calls if the provider adds `Allow-Control-*` headers blocking others domains.

**Comment by ivmarcos** [comment]
I removed the proxy disabling and just kept the change about the browser identification. Thank you all for the feedback :+1:!

**Comment by filipedeschamps** [comment]
Great discussion 👍 

Are we ready to merge this?

**Comment by filipedeschamps** [comment]
Something strange happened when I published the newest version `cep-promise@3.0.6` with this change. Before the module is published, it goes through the build process and it's generating different `dist` files for me... even if I'm currently on the HEAD of `master`.

Any idea? The impression I get is it's building the previous version:

image

**Comment by lucianopf** [comment]
Quick question @filipedeschamps , at this diff that you sent can you please check the one that actually changes the `typeof proccess === 'undefined'` to `isBrowser()`? 🤔 

Because if it's updated maybe there's a cached dependency at @ivmarcos computer (at a higher or lower version) and maybe that's affecting how `rollup` is bundling and precompiling our resources. 🤔 

(Still pretty confusing though)

**Comment by ivmarcos** [comment]
Sorry guys, I checked it here and seems that what @lucianopf  mentioned is true, I was using ```yarn``` and after reinstalling the ```node_modules``` using ```npm``` the build files are with the same green lines as posted above by @filipedeschamps. Anyway, I tested here ```3.3.6``` and it's nice, no cors errors! Thanks for the support :smiley:!

**Pull request merged**